include/mbedtls/check_config.h

```diff
@@ -962,7 +962,14 @@
 #error "MBEDTLS_SSL_VARIABLE_BUFFER_LENGTH defined, but not all prerequisites"
 #endif
 
+#if defined(MBEDTLS_SSL_SESSION_TICKETS) && \
+    !( defined(MBEDTLS_GCM_C) || defined(MBEDTLS_CCM_C) || defined(MBEDTLS_CHACHAPOLY_C) )
+#error "MBEDTLS_SSL_SESSION_TICKETS defined, but not all prerequisites"
```

Comment:
Hmm, sorry, can you remind me why `SSL_SESSION_TICKETS`, as opposed to `SSL_TICKET_C`, has a dependency on AEAD? I'm afraid we might have done the wrong thing here (and I'm afraid it was based on a suggestion from me, sorry if that's the case.)

Reply:
From the commit message:
> Both functions are calling mbedtls_cipher_auth_[encrypt/decrypt]_ext() functions. These functions are guarded with MBEDTLS_CIPHER_MODE_AEAD || MBEDTLS_NIST_KW_C flags - make it consistent.
As a result ssl_server2 won't build now with MBEDTLS_SSL_SESSION_TICKETS enabled (mbedtls_cipher_auth_[encrypt/decrypt]_ext() functions not available).
Mark MBEDTLS_SSL_SESSION_TICKETS as dependent on MBEDTLS_CIPHER_MODE_AEAD || MBEDTLS_NIST_KW_C and disable MBEDTLS_SSL_SESSION_TICKETS in stream cipher only build.

Then `MBEDTLS_CIPHER_MODE_AEAD` was switched to `defined(MBEDTLS_GCM_C) || defined(MBEDTLS_CCM_C) || defined(MBEDTLS_CHACHAPOLY_C)` as it couldn't be used in `check_config.h`.

Then `MBEDTLS_NIST_KW_C` was removed.

Reply:
- `mbedtls_cipher_auth_[encrypt/decrypt]ext()` functions depend on `AEAD (MBEDTLS_GCM_C || MBEDTLS_CCM_C || MBEDTLS_CHACHAPOLY_C)`
- These functions are used in `ssl_msg.c` (guarded with `MBEDTLS_GCM_C || MBEDTLS_CCM_C || MBEDTLS_CHACHAPOLY_C`) and `ssl_ticket.c`.
- As you mentioned [here](https://github.com/Mbed-TLS/mbedtls/pull/6374#discussion_r984300455) `SSL_SESSION_TICKETS` should depend on `AEAD` and `GCM || CCM || ChachaPoly`, but you refered to `SSL_TICKET_C` (not `SSL_SESSION_TICKETS`).
- We can't build with `MBEDTLS_SSL_SESSION_TICKETS` and without `SSL_TICKET_C`. So `MBEDTLS_SSL_SESSION_TICKETS` also depends on `GCM || CCM || ChachaPoly`.

Summary:
For me it should be fixed as follows:
- `SSL_TICKET_C` depends on `GCM || CCM || ChachaPoly`.
- `MBEDTLS_SSL_SESSION_TICKETS` depends on  `SSL_TICKET_C`.

Reply:
This approach seems right to me.

Reply:
Ok but in this case `ssl_server2.c` should use `MBEDTLS_SSL_SESSION_TICKETS && SSL_TICKET_C` guards. Linker undefined references:
```
ssl_server2.c:(.text.startup+0x17b): undefined reference to `mbedtls_ssl_ticket_init'
/usr/bin/ld: ssl_server2.c:(.text.startup+0x93d): undefined reference to `mbedtls_ssl_ticket_free'
/usr/bin/ld: ssl_server2.c:(.text.startup+0x1f1c): undefined reference to `mbedtls_ssl_ticket_setup'
/usr/bin/ld: ssl_server2.c:(.text.startup+0x1f32): undefined reference to `mbedtls_ssl_ticket_parse'
/usr/bin/ld: ssl_server2.c:(.text.startup+0x1f39): undefined reference to `mbedtls_ssl_ticket_write'
/usr/bin/ld: ssl_server2.c:(.text.startup+0x1fcd): undefined reference to `mbedtls_ssl_ticket_rotate'
```

Reply:
Yes, indeed I think the current guards are wrong and `MBEDTLS_SSL_SESSION_TICKETS && SSL_TICKET_C` would be correct.